tests/test_rank.py

```diff
@@ -403,42 +392,139 @@ def test_get_knn_distance_ood_scores():
-    # Get KNN distance as OOD score
+    # Get KNN distance as outlier score
     k = 5
-    knn_distance_ood_score = rank.get_knn_distance_ood_scores(
-        features=X_test_with_ood, nbrs=nbrs, k=k
-    )
+    knn_distance_to_score = rank.get_outlier_scores(features=X_test_with_ood, knn=knn, k=k)
 
-    # Index of the datapoint with max ood score should correspond to the last element of X_test_with_ood
-    idx_max_score = np.argsort(knn_distance_ood_score)[-1]
+    # Index of the datapoint with max ood score should correspond to the first element of X_test_with_ood
+    idx_max_score = np.argsort(knn_distance_to_score)[0]
     assert idx_max_score == (X_test_with_ood.shape[0] - 1)
 
-    # Get KNN distance as OOD score without passing k
-    # By default k=min(10, max_k) is used where max_k is extracted from the given nbrs
-    knn_distance_ood_score = rank.get_knn_distance_ood_scores(features=X_test_with_ood, nbrs=nbrs)
+    # Get KNN distance as outlier score without passing k
+    # By default k=10 is used or k = n_neighbors when k > n_neighbors extracted from the knn
+    knn_distance_to_score = rank.get_outlier_scores(features=X_test_with_ood, knn=knn)
+    # Index of the datapoint with max outlier score should correspond to the first element of X_test_with_ood
+    idx_max_score = np.argsort(knn_distance_to_score)[0]
+    assert idx_max_score == (X_test_with_ood.shape[0] - 1)
+
+    # Get KNN distance as outlier score passing k and t > 1
+    large_t_knn_distance_to_score = rank.get_outlier_scores(
+        features=X_test_with_ood, knn=knn, k=k, t=5
+    )
 
-    # Index of the datapoint with max ood score should correspond to the last element of X_test_with_ood
-    idx_max_score = np.argsort(knn_distance_ood_score)[-1]
+    # Index of the datapoint with max ood score should correspond to the first element of X_test_with_ood
+    idx_max_score = np.argsort(large_t_knn_distance_to_score)[0]
     assert idx_max_score == (X_test_with_ood.shape[0] - 1)
 
+    # Get KNN distance as outlier score passing k and t < 1
+    small_t_knn_distance_to_score = rank.get_outlier_scores(
+        features=X_test_with_ood, knn=knn, k=k, t=0.002
+    )
 
-def test_bad_k_for_get_knn_distance_ood_scores():
+    # Index of the datapoint with max ood score should correspond to the first element of X_test_with_ood
+    idx_max_score = np.argsort(small_t_knn_distance_to_score)[0]
+    assert idx_max_score == (X_test_with_ood.shape[0] - 1)
+    assert np.sum(small_t_knn_distance_to_score) >= np.sum(large_t_knn_distance_to_score)
 
-    X_train = data["X_train"]
-    X_test = data["X_test"]
 
-    # Create OOD datapoint
+@pytest.mark.filterwarnings("ignore::UserWarning")
+def test_default_k_and_model_get_outlier_scores():
+    # Testing using 'None' as model param and correct setting of default k as max_k
+
+    # Create dataset with OOD example
+    X = data["X_test"]
     X_ood = np.array([[999999999.0, 999999999.0]])
+    X_with_ood = np.vstack([X, X_ood])
```

Comment:
maybe I'm missing something but it seems none of the unit tests are explicitly verifying that X_ood has the smallest outlier score among all the datapoints? 

We should definitely explicitly verify that when `X_with_ood` is passed in, the final index of `outlier_scores` contains the smallest value in this vector.

Reply:
This was being checked using argsort but I made it more readable!

Reply:
Ah got it, looks fine sorry I missed it!